### PR TITLE
feat: Kai search results

### PR DESCRIPTION
**Pull request by richburdon**

### CODE REVIEW

File: packages/experimental/kai/src/containers/SearchPanel/SearchPanel.tsx [hunk 1]
```diff
@@ -159,8 +160,12 @@ export const SearchPanel = ({ onSelect }: SearchPanelProps) => {
...
 
   const { space } = useAppRouter();
-  const results = useQuery(space).map(matchFilter(text)).filter(Boolean) as SearchResult[];
-  const sorted = results.sort(sort);
+  const results = useQuery(space).map(matchFilter(text));
+  const sorted = (results.filter(Boolean) as SearchResult[]).sort(sort);
+  useEffect(() => {
+    // TODO(burdon): Causes recursion.
+    // onResults?.(sorted);
+  }, [results]); // TODO(burdon): The `sorted` array would always be updated.
```
Comment (dmaretskyi) [inline packages/experimental/kai/src/containers/SearchPanel/SearchPanel.tsx:168]:
```suggestion
  }, [...getIdsFromSelection(result)]); // TODO(burdon): The `sorted` array would always be updated.
```

File: packages/experimental/kai/src/containers/SearchPanel/SearchPanel.tsx [hunk 2]
```diff
@@ -159,8 +160,12 @@ export const SearchPanel = ({ onSelect }: SearchPanelProps) => {
   };
 
   const { space } = useAppRouter();
-  const results = useQuery(space).map(matchFilter(text)).filter(Boolean) as SearchResult[];
-  const sorted = results.sort(sort);
+  const results = useQuery(space).map(matchFilter(text));
```
Comment (dmaretskyi) [inline packages/experimental/kai/src/containers/SearchPanel/SearchPanel.tsx:163]:
```suggestion
  const results = useQuery(space, matchFilter(text), [text]);
```

File: packages/common/react-components/src/components/Dialog/Dialog.tsx [hunk 3]
```diff
@@ -89,6 +89,7 @@ export const Dialog = ({
           onCloseAutoFocus={(event) => event.preventDefault()}
           {...slots.content}
           className={mx(
+            'flex flex-col',
```
Comment (thure) [inline packages/common/react-components/src/components/Dialog/Dialog.tsx:92]: `flex-col` by default can cause very subtle bugs except in those very specific cases where the element has an extrinsic size and one of its children should grow to fill it. Only part of that scenario is present here, so there is no cause to have `flex-col`.

File: packages/common/react-components/src/components/Dialog/Dialog.tsx [hunk 4]
```diff
@@ -101,10 +102,11 @@ export const Dialog = ({
             <DialogPrimitive.Title
               {...slots.title}
               className={mx(
+                'shrink-0',
                 'text-xl font-system-medium text-neutral-900 dark:text-neutral-100 rounded-md',
                 titleVisuallyHidden && 'sr-only',
                 defaultFocus,
-                slots.content?.className
+                slots?.title?.className
```
Comment (thure) [inline packages/common/react-components/src/components/Dialog/Dialog.tsx:109]: Ah, thank you for fixing this

File: packages/common/react-components/src/components/Dialog/Dialog.tsx (hunk 3 above)
Comment (thure) [reply packages/common/react-components/src/components/Dialog/Dialog.tsx:92]: If the Dialog that Kai is consuming often has an extrinsic size and `flex-col`, kai can reduce classname repetition in slots by defining its own Dialog component that sets those defaults.

File: packages/common/react-components/src/components/Table/Table.tsx [hunk 5]
```diff
@@ -74,11 +74,14 @@ export const Table = <T extends {}>({ columns, data = [], slots = {}, onSelect,
         <div className='thead sticky top-0'>
           {headerGroups.map((headerGroup) => (
             // eslint-disable-next-line react/jsx-key
-            <div {...headerGroup.getHeaderGroupProps()} className='tr h-[2.5rem] border-b border-zinc-300'>
+            <div {...headerGroup.getHeaderGroupProps()} className='tr h-[2rem] border-b border-zinc-300'>
```
Comment (thure) [inline packages/common/react-components/src/components/Table/Table.tsx:77]: 2rem is 32px, which is one of the density sizes… should this use one of those or remain arbitrary?

`import { fineBlockSize } from '../../styles/density'`

**Pull request closed without merging**